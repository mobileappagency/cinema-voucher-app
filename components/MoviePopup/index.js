// @flow
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import {
  Animated,
  Image,
  View,
  Text,
  TouchableHighlight,
  TouchableWithoutFeedback
} from 'react-native'
import type { MovieModel } from '../../data/movies'

type Props = {
  movie: MovieModel,
  isOpen: boolean,
  onClose: Function
}

export default class MoviePoster extends Component<Props> {
  static propTypes = {
    movie: PropTypes.object.isRequired,
    isOpen: PropTypes.bool.isRequired,
    onClose: PropTypes.func.isRequired
  }

  render () {
    return (
      <View>
        <TouchableWithoutFeedback onPress={this.props.onClose}>
          <Animated.View />
        </TouchableWithoutFeedback>
        <Animated.View>
          <View>
            <View>
              <View>
                <Image />
              </View>
              <View>
                <Text />
                <Text />
              </View>
            </View>

            <View>
              <Text>Day</Text>
              <Text>Add day options here</Text>
              <Text>Showtime</Text>
              <Text>Add show time options here</Text>
            </View>
          </View>
          <View>
            <TouchableHighlight underlayColor='#9575CD'>
              <Text>Book My Tickets</Text>
            </TouchableHighlight>
          </View>
        </Animated.View>
      </View>
    )
  }
}
