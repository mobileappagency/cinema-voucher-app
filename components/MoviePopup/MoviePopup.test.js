// @flow
import React from 'react'
import MoviePopup from './'
import { shallow } from 'enzyme'
import type { TvShowsResult } from 'types'

describe('MoviePopup', () => {
  let movie: TvShowsResult

  beforeEach(() => {
    movie = {
      genre_ids: [],
      name: 'Blade Runner',
      poster_path: 'http://movie-posters.com/blade-runner.jpg',
      overview: '... text ...',
      vote_average: 4.5,
      vote_count: 100000,
      original_language: 'en-US'
    }
  })

  it('matches current snapshot', () => {
    const rendered = shallow(<MoviePopup movie={movie} isOpen onClose={jest.fn()} />)

    expect(rendered).toMatchSnapshot()
  })

  it.skip('executed onClose', () => {
    const onCloseMock = jest.fn()

    const rendered = shallow(<MoviePopup movie={movie} isOpen onClose={onCloseMock} />)

    rendered.find('TouchableWithoutFeedback').simulate('press')

    expect(onCloseMock).toHaveBeenCalled()
  })
})
