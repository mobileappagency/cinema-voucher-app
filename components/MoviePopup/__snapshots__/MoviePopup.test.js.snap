// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePopup matches current snapshot 1`] = `
<View>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <AnimatedComponent />
  </TouchableWithoutFeedback>
  <AnimatedComponent>
    <View>
      <View>
        <View>
          <Image />
        </View>
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
        </View>
      </View>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Day
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Add day options here
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Showtime
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Add show time options here
        </Text>
      </View>
    </View>
    <View>
      <TouchableHighlight
        activeOpacity={0.85}
        underlayColor="#9575CD"
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Book My Tickets
        </Text>
      </TouchableHighlight>
    </View>
  </AnimatedComponent>
</View>
`;
